src/cc/api/BPFTable.cc

```diff
@@ -142,26 +142,37 @@ StatusTuple BPFTable::remove_value(const std::string& key_str) {
   return StatusTuple(0);
 }
 
+StatusTuple BPFTable::clear_table_non_atomic() {
```

Comment:
The implementation here takes advantage of kernel internal implementation that if a key is not found the kernel will search from the start. I think we should still stick to the public interface get_next and not relying on this particular kernel behavior.

Also, if implemented this function at BPFTable level, it should work for certain types of array maps as well.
For example, prog_array does support get_next and delete_elem. Using get_first and remove mechanism won't work for prog_array map. The same holds true for cgroup_fd_array and array_of_maps.

Reply:
Good point for the table types, will add check.

For the implementation, even if we use get_next, we still need to somehow find the first key to "get next for", right? If that's the case we could just keep re-using it anyways...

Reply:
Yes, for non-array map type, if the key cannot be found, it will find the first key, the same as you pass an invalid key or NULL pointer key.